src/kernels/jupyter/launcher/serverConnectionType.ts

```diff
@@ -1,41 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-import { inject, injectable, named } from 'inversify';
-import { EventEmitter, Memento } from 'vscode';
-import { GLOBAL_MEMENTO, IMemento, IsWebExtension } from '../../../platform/common/types';
-
-export const mementoKeyToIndicateIfConnectingToLocalKernelsOnly = 'connectToLocalKernelsOnly';
-
-@injectable()
-export class ServerConnectionType {
```

Comment:
I deleted this class and moved its functionality to the JupyterServerUriStorage class.

The reason for this is that this class was essentially exposing the internal state of the JupyterServerUriStorage class and they had to be in sync. And they weren't always.

To make this impossible, I just removed this class altogether and added a new interface to the JupyterServerUriStorage class. Most of the changes for this is simply using an interface instead of a class.

Good use case for why we should expose interfaces to the DI container and not classes.